apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -235,7 +235,7 @@ func (m *ModelManagerAPI) checkAddModelPermission(ctx context.Context, cloud str
 		Key:        cloud,
 	}
 	perm, err := m.accessService.ReadUserAccessLevelForTarget(ctx, user.NameFromTag(userTag), target)
-	if err != nil && !errors.Is(err, errors.NotFound) {
+	if err != nil && !errors.Is(err, accesserrors.AccessNotFound) {
```

Comment:
Yea, I wish Go had proper type unions, then you could maybe have an error return type that could only be certain error types

Reply:
Yes it would be interesting, but I am not sure it would be a very "golang" way to it (i mean, through type assert on error).

I wonder if we are not missing a point with the package level error definition, when we combo it with the duck typing idiom of golang.

Maybe we should take advantage of function like `errors.Join` and more generally wrapping error.

For instance, we could have few core errors (like not found), and assert that errors like XXXNotFound wrap it up ? 

It would make it possible to rely on general "notFound" concept in ducktyping whatever interface, but still be able to rely on specific error kind if we really need it.

I don't know: it bugs me :black_cat: